src/utils/modules.js

```diff
@@ -337,26 +334,31 @@ export class ModuleTracer extends Tracer {
       // Here, we handle calls of the form
       //   __d(factoryFunction, moduleId, dependencyArray)
 
-      if (this.evaluateForEffectsNesting !== 0)
-        this.modules.logger.logWarning(F, "Defining a module in nested partial evaluation is not supported.");
-      else {
-        let factoryFunction = argumentsList[0];
-        if (factoryFunction instanceof FunctionValue) {
-          let dependencies = this._tryExtractDependencies(argumentsList[2]);
-          if (dependencies !== undefined) this.modules.factoryFunctionDependencies.set(factoryFunction, dependencies);
-          else
-            this.modules.logger.logError(
-              argumentsList[2],
-              "Third argument to define function is present but not a concrete array."
-            );
+      let factoryFunction = argumentsList[0];
+      if (factoryFunction instanceof FunctionValue) {
+        let dependencies = this._tryExtractDependencies(argumentsList[2]);
+        if (dependencies !== undefined) {
+          let previousDependencies = this.modules.factoryFunctionDependencies.get(factoryFunction);
+          if (previousDependencies) {
+            // Verify that they are the same
+            invariant(previousDependencies.length === dependencies.length);
```

Comment:
If they are not the same (length / content), then I'd rather emit a logError than fail an invariant.